src/exo/API.py

```diff
@@ -696,17 +672,19 @@ def fuse_if(self, if1, if2):
 
         return Procedure(loopir, _provenance_eq_Procedure=self)
 
-    def add_loop(self, stmt, var, hi):
+    def add_loop(self, stmt, var, hi, guard=False):
```

Comment:
At every call site, we're making sure to write `guard=True/False`. We should consider _requiring_ the use of the keyword by changing the signature as follows:

```suggestion
    def add_loop(self, stmt, var, hi, *, guard=False):
```
Arguments after the `*` are _keyword-only_.